### PR TITLE
util/lru: update c.head when deleting the most recently used entry

### PR DESCRIPTION
**Pull request by danderson**
Fixes tailscale/corp#14747

Signed-off-by: David Anderson <danderson@tailscale.com>
Co-authored-by: Brad Fitzpatrick <bradfitz@tailscale.com>

### CODE REVIEW

File: util/lru/lru_test.go [hunk 1]
```diff
@@ -10,6 +10,7 @@ import (
 	"testing"
 
 	"github.com/google/go-cmp/cmp"
+	"golang.org/x/exp/maps"
```
Comment (bradfitz) [inline util/lru/lru_test.go:13]: elsewhere we've been importing this now as xmaps:

```
% git grep -F 'xmaps "'
cmd/tailscale/cli/exitnode.go:  xmaps "golang.org/x/exp/maps"
cmd/testwrapper/testwrapper.go: xmaps "golang.org/x/exp/maps"
ipn/ipnlocal/local.go:  xmaps "golang.org/x/exp/maps"
wgengine/filter/filter_test.go: xmaps "golang.org/x/exp/maps"
wgengine/magicsock/endpoint.go: xmaps "golang.org/x/exp/maps"
wgengine/magicsock/magicsock_test.go:   xmaps "golang.org/x/exp/maps"
```

(to distinguish it from the std maps)

File: util/lru/lru_test.go [hunk 2]
```diff
@@ -48,6 +49,156 @@ func TestLRU(t *testing.T) {
 	}
 }
 
+func TestLRUDeleteCorruption(t *testing.T) {
+	// Regression test for tailscale/corp#14747
+
+	c := Cache[int, bool]{}
+
+	c.Set(1, true)
+	c.Set(2, true)
```
Comment (bradfitz) [inline util/lru/lru_test.go:58]: c.Set(2, true) // now 2 is the head
c.Delete(2) // delete the head

**Review by bradfitz — APPROVED** [review]

File: util/lru/lru_test.go (hunk 1 above)
Comment (danderson) [reply util/lru/lru_test.go]: Done

File: util/lru/lru_test.go (hunk 2 above)
Comment (danderson) [reply util/lru/lru_test.go]: Done

**Pull request merged**